gelf/reader_test.go

```diff
@@ -0,0 +1,184 @@
+package gelf
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"net"
+	"strings"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	kSourceCount  = 10
+	kMsgPerSource = 100
+	kMaxMsgLength = 3000
+	kChunkSplit   = 1024
+
+	kSleep = 1
+)
```

Comment:
I would love to see more test cases slightly at the edge of the implementation;

- Can a client with a different source IP/port continue to assemble someone else's message?
- What happens with to short messages starting with the `magicChunked`

Reply:
> What happens with to short messages starting with the magicChunked
covered it with the test

> Can a client with a different source IP/port continue to assemble someone else's message? 

I do not see any limitations in GELF specification, so I assume that it's possible and allowed )